File: integration_testing/test_sap/sonar_client/test_client/sonar_client.py
import logging
from .http_client_base import CommonClient


class SonarClient(CommonClient):
    def __init__(self,
                 base_url='http://localhost:9000/',
                 auth=('', ''),
                 polling_threads=5,
                 polling_sleep=0.5,
                 polling_long_sleep=30,
                 polling_max_attempts=400,
                 logging_level=logging.WARNING):
        """
        Creates a new instance of a client object to access Sonarqube
        """
        logger = logging.getLogger('SonarqubeApiClient')
        logger.setLevel(logging_level)
        CommonClient.__init__(self,
                              base_url=base_url,
                              polling_threads=polling_threads,
                              polling_sleep=polling_sleep,
                              polling_long_sleep=polling_long_sleep,
                              polling_max_attempts=polling_max_attempts,
                              logging_level=logging_level)
        self.logger = logger

    def get_project_coverage(project):
        res = self.session.get(self.base_url + 
        'api/measures/search_history?component={project}&metrics=coverage'.format(project=project),
        verify=False, auth=self.auth)
        return res

File: integration_testing/test_axios/sonar_client/test_client/sonar_client.js
const CommonClient = require('./common_client')
const Constants = require('../constants')

class SonarClient extends CommonClient {
    constructor(baseUrl=Constants.baseUrl, auth=Constants.token) {
        super(baseUrl, auth)
        new CommonClient(this.baseUrl=baseUrl, this.auth=auth)
    }

    async get_metric_badge(project) {
        const url = Constants.PROJECT_BADGE_MEASURE_ENDPOINT
        return await this.session.get(url, {params: {project: project, metric: 'coverage'}})
    }

    async get_coverage(project) {
        const url = Constants.PROJECT_METRICS_ENDPOINT
        return await this.session.get(url, {params: {component: project, metrics: 'coverage'}})
    }
}

module.exports = SonarClient
